service/history/task/task.go

```diff
@@ -277,13 +277,20 @@ func (t *taskImpl) HandleErr(
 		return nil
 	}
 
-	if transferTask, ok := t.Info.(*persistence.TransferTaskInfo); ok &&
-		transferTask.TaskType == persistence.TransferTaskTypeCloseExecution &&
-		err == execution.ErrMissingWorkflowStartEvent &&
-		t.shard.GetConfig().EnableDropStuckTaskByDomainID(t.Info.GetDomainID()) { // use domainID here to avoid accessing domainCache
-		t.scope.IncCounter(metrics.TransferTaskMissingEventCounterPerDomain)
-		t.logger.Error("Drop close execution transfer task due to corrupted workflow history", tag.Error(err), tag.LifeCycleProcessingFailed)
-		return nil
+	if err == execution.ErrMissingWorkflowStartEvent {
+		if t.shouldDropTaskOnError() {
+			t.scope.IncCounter(metrics.TaskMissingEventCounterPerDomain)
+			t.logger.Error("Drop task due to corrupted workflow history", tag.Error(err), tag.TaskType(t.GetTaskType()), tag.LifeCycleProcessingFailed)
+			return nil
+		}
+	}
+
+	if execution.IsConflictError(err) {
```

Comment:
I don't think it's safe to drop a task based on a conflict error, which can happen for non-corruption reasons.

Reply:
even if we have retried multiple times?